pkg/asset/manifests/openshift.go

```diff
@@ -152,9 +153,12 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 			},
 		}
 	case vspheretypes.Name:
+		// Strip illegal colons which may apppear if VCenter has a port.
+		vCenter := strings.Split(installConfig.Config.VSphere.VCenter, ":")[0]
```

Comment:
hmm.. what if the VCenter location is a IPv6 address?

Reply:
or what if the vcenter location is a HTTPs url? _ not sure if that's valid in the first place_

Reply:
> hmm.. what if the VCenter location is a IPv6 address?

The certificates provided by vcenter do not include either versions of ip addressing.  They should really only be providing FQDN.

Reply:
> or what if the vcenter location is a HTTPs url? _ not sure if that's valid in the first place_

govmomi handles if the string is either a url or not.  We probably be explict that it is just a FQDN.

Reply:
> hmm.. what if the VCenter location is a IPv6 address?

An IPv6 address does not seem acceptable based on the upstream constraint...

> or what if the vcenter location is a HTTPs url? _ not sure if that's valid in the first place_

I just tested and a transport like https:// will be passed through, so this is another option that would break us. I'm not sure if we should just block on validation with transport or filter for the user. Probably block... they should get their install config right.

> The certificates provided by vcenter do not include either versions of ip addressing. They should really only be providing FQDN.

Accepting only FQDN would be simplest but it seems like a lot of people are using IP addresses?

Reply:
what about just replacing all non-letter characters with underscores?